Backend/Remora.Discord.Caching/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -48,16 +50,33 @@ public static class ServiceCollectionExtensions
     /// <remarks>
     /// The cache uses a run-of-the-mill <see cref="IMemoryCache"/>. Cache entry options for any cached type can be
     /// configured using <see cref="IOptions{CacheSettings}"/>.
+    ///
+    /// When choosing a cache implementation, it should be noted that <see cref="AddDiscordCaching"/> and
+    /// <see cref="AddDiscordRedisCaching"/> cannot be used together.
     /// </remarks>
     /// <param name="services">The services.</param>
     /// <returns>The services, with caching enabled.</returns>
     public static IServiceCollection AddDiscordCaching(this IServiceCollection services)
     {
-        services
-            .AddMemoryCache();
+        services.AddMemoryCache();
+
+        services.TryAddSingleton<ICacheService, CacheService>();
+
+        services.AddCachingAPIAndResponders();
 
+        return services;
+    }
+
+    /// <summary>
+    /// Adds a caching implementations of various Discord API types and caching responders to the service collection.
+    /// </summary>
+    /// <param name="services">The service collection to add the caching APIs to.</param>
+    /// <remarks>Invoking this method alone is not recommended as an implementation of
+    /// <inheritdoc cref="ICacheService"/> must also be provided. Consider using <see cref="AddDiscordCaching"/> instead.</remarks>
+    /// <returns>The service collection to chain calls with.</returns>
+    public static IServiceCollection AddCachingAPIAndResponders(this IServiceCollection services)
```

Comment:
Move the responder adding into `AddDiscordCaching` (since those always need to be included when caching is enabled) and rename this to just `AddCachingAPI`. There could probably be a case made for not having this method at all, and just having it all in `AddDiscordCaching`, too.

Reply:
Well I considered this, but `AddDiscordRedisCaching` requires them too, but then you have a potentially needless `IMemoryCache` added to your container. I'd be fine with reverting this change though.